core/driver/sources/src/test/scala/org/virtuslab/ideprobe/dependencies/IntelliJCleanupTest.scala

```diff
@@ -0,0 +1,76 @@
+package org.virtuslab.ideprobe.dependencies
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.virtuslab.ideprobe.config.{DependenciesConfig, IdeProbeConfig, IntelliJProvider, IntellijConfig, PathsConfig}
+import org.virtuslab.ideprobe.ide.intellij.IntelliJFactory
+
+import java.nio.file.Path
+
+@RunWith(classOf[JUnit4])
+final class IntelliJCleanupTest {
+  @Test
+  def existingIntelliJShouldNotBeDeletedDuringCleanup: Unit = givenInstalledIntelliJ { installationRoot =>
+      val intelliJProvider = IntelliJProvider.from(
+        IdeProbeConfig(
+          intellij = IntellijConfig.Existing(installationRoot, Seq.empty),
+          workspace = None,
+          resolvers = DependenciesConfig.Resolvers(),
+          driver = IntelliJFactory.Default.config,
+          paths = PathsConfig()
+        )
+      )
+      val existingInstalledIntelliJ = intelliJProvider.setup()
+
+      //when
+      existingInstalledIntelliJ.cleanup()
+
+      //then
+      assert(installationRoot.toFile.exists(), "The provided IntelliJ instance should exist after cleanup, but it was deleted.")
+    }
+
+  @Test
+  def existingIntelliJShouldRetainItsOriginalPluginsDuringCleanup: Unit = givenInstalledIntelliJ { installationRoot =>
+    //given a pre-installed IntelliJ and an IntelliJProvider
+    val preInstalledPlugins = installationRoot.resolve("plugins").toFile.list().toSet
+    val pluginsToInstall = Seq( //Plugins choice is arbitrary.
```

Comment:
this looks unreadable and might change with IJ upgrade. We rather keep our tests using latest version of IJ to make sure it keeps working.
Perhaps you should install some known plugin, like scala plugin in original intellij, then for example install some other plugin or a plugin in different version and then check if scala plugin is back after cleanup.

Reply:
I opted for these plugins as it is highly improbable that they will be pre-installed. I'll fix it together with the factory/provider renaming.

Reply:
Changed to Scala plugin (together with creating the provider from config).